### DISCUSSION TITLE
Get BitDepth

### DISCUSSION THREAD

**Discussion by endasil** [Zeugma440/atldotnet#173] [Q&A] [answered]
I would like to get the bit depth of an audio file but there does not seem to be a property for that, is there a way to get it?

**Comment by Zeugma440** [comment]
Could your question be a duplicate of #119 by any chance?

**Comment by endasil** [comment]
I tried to calculate bit depth using that method but got wrong values.  I tried with a file with a bit depth of 24.
It has a BitRate of 2117, a sample rate of 44100.

1000 * Track.Bitrate / Track.SampleRate => 1000 * 2117 * 44100 = 48.004535147392289 
Googling the topic now seem to suggest bits per sample and bit depth are the same like you said. Now after some more googling I found out that you should also take the number of channels into account to get the correct value.

ATL.Track audioInfo = new Track(pathToFile);
var bitRate = (int) ((audioInfo.Bitrate * 1024) / audioInfo.SampleRate /
                                  audioInfo.ChannelsArrangement.NbChannels);

I also needed to change 1000 to 1024 otherwise I get 31 instead of 32 on 32 bit tracks.

**Reply by Zeugma440** [reply]
You're completely right : each sample is taken for one given channel, therefore the number of channels should be taken into account. My apologies.

Bitrate uses 1000 and not 1024 : http://www.lyberty.com/encyc/articles/kb_kilobytes.html

Looks like you should use 1000 when measuring data streams and 1024 when measuring data storage.

**Reply by endasil** [reply]
That's interesting, what could give me a result of 31 instead of 32 bits for 32 bit audio then? Is there something else i missed?

**Reply by Zeugma440** [reply]
> what could give me a result of 31 instead of 32 bits for 32 bit audio then? Is there something else i missed?

Algorithm-wise, you haven't missed anything else. However, I can't help much more without having a look at the file. I'd blame rounding issues for starters 😉

**Reply by endasil** [reply]
Here is a file i'm having problem with. https://drive.google.com/file/d/1ybMx_6ksfh69o3rwtqhJ1D31d4s75FXV/view?usp=share_link

When I check the Track.BitRate on it with the atl lib I get 416. When looking at the file in windows explorer it says 449 which does seem a bit strange to me. Perhaps this could a reason for the unexpected rounding error above?

Regardless of which I use i get the wrong bit depth.

Bitrate(449) * 1000 = 449 000
BitRateInBytes(449 000) / SampleRate(44100)  / 2 =  5  


File explorer says this is a 16 bit file and I get 5. How come?

**Reply by Zeugma440** [reply]
Alright. I've created a bona fide `Track.BitDepth` property that will make things way easier for you.

That new field will be available on next release.

**Reply by endasil** [reply]
Nice! Thank you! When do you think you will have a new release out? I'm excited to try it out. :)

**Reply by Zeugma440** [reply]
Will do between today and tomorrow

**Comment by Zeugma440** [comment, marked as answer]
BIt depth is now accessible through `Track.BitDepth` in today's v4.15
